pallets/identity/src/lib.rs

```diff
@@ -590,23 +590,21 @@ decl_module! {
         pub fn set_permission_to_signer(
             origin,
             signer: Signatory<T::AccountId>,
-            permissions: secondary_key::api::Permissions
+            permissions: Permissions
         ) -> DispatchResult {
-            let PermissionedCallOriginData {
-                sender,
-                primary_did: did,
-                ..
-            } = Self::ensure_origin_call_permissions(origin)?;
-            let record = Self::grant_check_only_primary_key(&sender, did)?;
+            Self::base_set_permission_to_signer(origin, signer, permissions)
+        }
 
-            // You are trying to add a permission to did's primary key. It is not needed.
-            match signer {
-                Signatory::Account(ref key) if record.primary_key == *key => Ok(()),
-                _ if record.secondary_keys.iter().any(|si| si.signer == signer) => {
-                    Self::update_secondary_key_permissions(did, &signer, permissions.into())
-                }
-                _ => Err(Error::<T>::InvalidSender.into()),
-            }
+        /// This function is a workaround for https://github.com/polkadot-js/apps/issues/3632
+        /// It sets permissions for an specific `target_key` key.
+        /// Only the primary key of an identity is able to set secondary key permissions.
+        #[weight = <T as Trait>::WeightInfo::set_permission_to_signer()]
+        pub fn legacy_set_permission_to_signer(
+            origin,
+            signer: Signatory<T::AccountId>,
+            permissions: secondary_key::api::LegacyPermissions
+        ) -> DispatchResult {
+            Self::base_set_permission_to_signer(origin, signer, permissions.into())
```

Comment:
You could just call `set_permission_to_signer` and move the contents of the `base_` function there.
```suggestion
            Self::set_permission_to_signer(origin, signer, permissions.into())
```